mu-plugins/blocks/google-map/index.php

```diff
@@ -87,11 +89,65 @@ function render( $attributes, $content, $block ) {
 
 	?>
 
-	<div <?php echo wp_kses_data( $wrapper_attributes ); ?>>
-		Loading...
+	<div class="wporg-google-map-container">
+		<?php if ( $attributes['showFilters'] ) : ?>
+			<?php echo do_blocks( '
+				<!-- wp:group {"style":{"spacing":{"blockGap":"0"}},"layout":{"type":"flex","flexWrap":"nowrap"},"className":"wporg-google-map-query-filters"} -->
+					<div class="wp-block-group wporg-query-filters">
+						<!-- wp:wporg/query-filter {"key":"map_format"} /-->
+						<!-- wp:wporg/query-filter {"key":"map_type"} /-->
+						<!-- wp:wporg/query-filter {"key":"map_month"} /-->
+						<!-- wp:wporg/query-filter {"key":"map_country"} /-->
+					</div>
+				<!-- /wp:group -->
+			' ); ?>
+		<?php endif; ?>
+
+		<!-- TODO: rearrange this so that .wporg-google-map-query-filters above and the map/list/search elements below are all direct descendents of the same container.
+		  -- then use Grid to change the `order` so that filters shows up after search
+		  -->
```

Comment:
I'm not seeing a good way to do this. If react renders into `wporg-google-map-container` then it'll wipe out the filters. In order to hydrate instead of render, the server markup would have to match the react markup. [Subgrid support isn't good enough yet](https://caniuse.com/?search=subgrid).

 I could do something hacky, like using JS to save the `innerHTML` content to a variable, set `display:none`, then  use `<RawHTML>` (`dangerouslySetInnerHTML`) to render it inside the react container. It feels like there should be a better way, though

Reply:
I also thought about refactoring so that each React component is independent. So, the server would generate something like this:

```html
<div id="container">
	<div id="map"></div>
	<div id="search"></div>

	<div id="filters">
		<?php echo do_blocks( <!-- wp:wporg/query-filter ... --> ); ?>
	</div>

	<div id="list"></div>
</div>
```

...then call `createRoot()` for `map`, `search`, and `list` seperately. The problem there is that there needs to be a `Main` component managing the state for those subcomponents.


Another idea might be to basically re-create the `InnerBlocks` / `ServerSideRender` concept, and grab the `query-filter` content via a REST API request, then render the result into the DOM. That's similar to the hacky idea about about grabbing the `InnerHTML`, though 🤷🏻‍♂️